From edd4fd50792bfc84df31c05b556ec0a15be2400e Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Fri, 29 Oct 2021 15:39:04 +0300
Subject: [PATCH] [SW-0] gaudi2: arc: set internal dccm address

Currently, stack pointer of ARC0 which runs pcie initialization
code is set DCCM using HBW address. Since ARC access to DCCM via
HBW address is slower compared to direct address access, this patch
uses direct address to set stack pointer of ARC0.

Change-Id: I07a4ebe707fb9778a18be112be81c8efdaf181c6
---

diff --git a/arch/gaudi2/arc/secondary.S b/arch/gaudi2/arc/secondary.S
index 0ea094d..2c0e567 100644
--- a/arch/gaudi2/arc/secondary.S
+++ b/arch/gaudi2/arc/secondary.S
@@ -7,7 +7,8 @@
 
 .global secondary_entry
 secondary_entry:
-	mov	sp, 0x27020400	/* stack top set to secondary CPU DCCM 1k offset */
+	/* stack top set to secondary CPU DCCM 1k offset (internal address) */
+	mov	sp, 0x80000400
 	mov	fp, sp
 
 	/* Disable un-aligned checks on data memory */
